ref/general/observability-monitoring.adoc

```diff
@@ -0,0 +1,65 @@
+
+== Metrics
+
+Metrics are numerical measurements that are captured and processed to identify and respond to potential issues before they escalate and cause more serious problems.
+In distributed system metrics, three are factors are involved; producers, aggregators, and processors.
+
+Producers, for example, applications, can be directly involved in producing metrics. Also, other infrastructure might passively observe or actively instrument the application to produce metrics on its behalf.
+What happens next depends on the aggregator.
+
+In the case of aggregators, for example, Prometheus, metrics are transferred from the producer to the aggregator by way of either `push` or `pull` mechanisms.
+Some aggregators, expect the application to connect to the aggregator to transmit data.
+The connection information for the aggregator must be distributed to all of the application processes that are measured.
+Other aggregators periodically connect to a known endpoint to gather metrics data.
+This requires the producer to define and provide an endpoint that can be scraped, and for the aggregator to be told where the endpoints are.
+When used with Kubernetes, Prometheus can discover endpoints based on service annotations.
+
+Processors, for example, Grafana, uses all of the aggregated data.
+Services like Grafana process aggregated metrics for visualization by using dashboards.
+Grafana also supports alerts by rules that are stored and evaluated independently of the dashboard.
+
+== Logs
+
+Log messages are strings with contextual information that pertains to the state and activity of a microservice when the log entry is created.
+Logs are required to diagnose how and why services fail. They play a supporting role to metrics in monitoring application health.
+Handling log files requires more thought if a containerized application can't be configured to write logs to `standard out` or `standard err`.
+You can use a volume for log data with a simple bind mount for local development and test, or a proper Persistent Volume as part of a Kubernetes deployment.
+A dedicated sidecar or logging agent can read from a shared volume to forward logs to a centralized aggregator.
+Log rotation must be configured explicitly to control the amount of log data stored on volumes.
+Alternately, you can use application libraries or agents to forward logs directly to aggregators.
+This option can carry some configuration complexity across deployment environments.
+
+== Tracing
+
+Distributed, request, or end-to-end tracing captures the end-to-end flow of a request through the system.
+Distributed tracing is a technique that addresses logging information in microservice-based applications.
+HTTP request tracing is a feature of IIS that provides a way to determine what exactly is happening with a request.
+This includes any form of authentication, which handler was used, and how long each step took in the pipeline.
```

Comment:
Change "which handler was used, and how long each step took in the pipeline" to "the choice of handler, and the length of each step that was taken in the pipeline" to match the syntax of the first part of the clause.